xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-script/src/main/java/org/xwiki/security/authentication/script/AuthenticationScriptService.java

```diff
@@ -121,4 +140,94 @@ public void resetAuthenticationFailureCounter(String username)
     {
         this.authenticationFailureManager.resetAuthenticationFailureCounter(username);
     }
+
+    /**
+     * Compute a relative URL for an {@link AuthenticationResourceReference} based on the given action string.
+     * See {@link AuthenticationResourceReference.AuthenticationAction} for more information.
+     *
+     * @param action the authentication action from which to build the right URL.
+     * @param params the query string parameters of the URL.
+     * @return a relative URL for the current wiki.
```

Comment:
apparently it can return null, this needs to be explained.

Reply:
Right